packages/devtools/lib/src/timeline/timeline.dart

```diff
@@ -275,22 +295,29 @@ class TimelineScreen extends Screen {
     }
   }
 
-  /// Adds buttons to the timeline that will dump debug information to text
-  /// files and download them. This will only appear if either the
-  /// [debugTimeline] flag is true or the [debugCpuProfile] flag is true.
+  void _exportTimeline() {
+    // There will be a trailing comma on [debugTraceEvents]. Remove so that we
+    // do not have to deal with json formatting errors later.
+    final traceEvents = debugTraceEvents.isEmpty
+        ? ''
+        : debugTraceEvents.toString().replaceRange(
+            debugTraceEvents.length - 1, debugTraceEvents.length, '');
+    final String json = '{'
+        '"traceEvents":[$traceEvents],'
+        '"cpuProfile":[${debugCpuProfileResponse.toString()}]}';
+    downloadFile(json, 'timeline_${DateTime.now().toString()}.json');
+  }
+
+  /// Adds a button to the timeline that will dump debug information to text
+  /// files and download them. This will only appear if the [debugTimeline] flag
+  /// is true.
   void _maybeAddDebugButtons() {
     if (debugTimeline) {
       debugButtonSection.add(PButton('Debug dump timeline')
         ..small()
         ..click(() {
-          // Trace event json in the order we received the events.
-          String traceEvents = debugTraceEvents.toString();
-          traceEvents = traceEvents.replaceRange(
-              traceEvents.length - 1, traceEvents.length, ']}');
-          downloadFile(traceEvents, 'trace_output.json');
-
           // Trace event json in the order we handled the events.
-          String handledTraceEvents = debugTraceEvents.toString();
+          String handledTraceEvents = debugHandledTraceEvents.toString();
           handledTraceEvents = handledTraceEvents.replaceRange(
```

Comment:
to switch over to encoding a large nested blob of JSON you will want to remove this toString call as well.